node/caching-compiler.js

```diff
@@ -84,15 +86,27 @@ function readWithCache(info, content) {
   info.cache[cacheFilename] = content;
 
   if (typeof info.cacheDir === "string") {
-    // Writing cache files is something that should only happen at package
-    // development time, so it's acceptable to use fs.writeFileSync
-    // instead of some complicated asynchronous-but-atomic strategy.
-    fs.writeFileSync(absCachePath, content, "utf8");
+    scheduleWrite(absCachePath, content);
   }
 
   return content;
 }
 
+const pendingWrites = Object.create(null);
+let pendingWriteTimer;
+
+function scheduleWrite(path, content) {
+  pendingWrites[path] = content;
+  pendingWriteTimer = pendingWriteTimer || setTimeout(function () {
+    pendingWriteTimer = null;
+    Object.keys(pendingWrites).forEach(function (path) {
+      const content = pendingWrites[path];
+      delete pendingWrites[path];
+      fs.writeFileSync(path, content, "utf8");
+    });
+  }, 10);
+}
```

Comment:
> It's a hash of file contents and the "reify" options from package.json

Awesome, I was hoping that was it.

> Is there some other reason for using `setImmediate`?

If there's no compat concern it's the "more correct" way to defer an operation is all.